doc/user/content/overview/architecture.md

```diff
@@ -8,23 +8,124 @@ menu:
+
+Each cluster replica is essentially a clone, constructing the same indexes and materialized views. [TODO: is this also true of ad-hoc select queries?]. Each cluster replica receives a copy of all data that comes in from sources its dataflows use, and uses the data to perform identical computations. This design provides Materialize with active replication, and so long as one replica is still reachable, the cluster continues making progress.
+
+[todo: operational concerns for cluster replicas]
+todo: you can add more replicas (to be more crash tolerant)
+todo: you can scale replica sizes up and down (to be more performant)
+todo: are there weird wrinkles here? like if I have 1 xl replica and one small replica my expectation is that i will get the speed of the xl with the fault tolerance associated with having a second replica, or are there cases where my speed will be bottlenecked by the small?
+
+#### Indexes
+
+Indexes assemble and maintain a a source, unmaterialized or materialized view's data in memory within all cluster replicas associated with a given cluster. Users can create indexes on any source, unmaterialized view, or materialized view present in the region. This enables future `SELECT` statements against that source, unmaterialized or materialized view to complete much faster, because the data can be directly read from in memory from a cluster replica rather than needing to be read from storage.
+
+Indexes are associated with a particular cluster, and indexes are replicated on all cluster replicas for that cluster. This setup lets us decouple changes to the computation (which views are defined, which indexes are created) from how that computation is physically manifested (how many replicas, how large is each replica). Note however, that unlike materialized or unmaterialized views, indexes are local to a cluster and can can only be used by other indexes or `SELECT` queries on the same cluster.
+
+For a deeper dive into how indexes work see [todo].
+
+#### Materialized views
+
+Materialized views take any query and incrementally maintain it, but instead of keeping the results in memory like indexes, the results are written back to persistent storage. `SELECT` queries against a materialized view result in reads from the storage layer, so if you need lower latency we recommend creating indexes based on common query patterns.
+
+Materialized views are defined on a particular cluster that is tasked with doing the required computations, but they can be read from any cluster in the region.
+
+### SQL Control Plan (Adapter)
+
+The SQL control plane is responsible for handling users input SQL statements, and assigning times to users read (`SELECT`, `SUBSCRIBE`, `COPY ... FROM`) queries and write (`INSERT`, `UPDATE`, `DELETE`, `COPY ... TO`) queries. The SQL control plane is also responsible for ensuring that transactions work correctly.
```

Comment:
Can link to [`/overview/isolation-level/`](https://materialize.com/docs/overview/isolation-level/).